src/box/lua/net_box.c

```diff
@@ -2563,7 +2574,8 @@ netbox_transport_process_requests(struct netbox_transport *transport,
 		if (netbox_transport_send_and_recv(transport, &hdr) != 0)
 			luaT_error(L);
 		netbox_transport_dispatch_response(transport, L, &hdr);
-		if (hdr.schema_version > 0 &&
+		if (transport->opts.fetch_schema == true &&
+			hdr.schema_version > 0 &&
```

Comment:
You don't need to check `fetch_schema` here: the function will go return the new `schema_version`, then go to `netbox_transport_fetch_schema`, which will return the new `schema_version` without doing anything, then go back here and continue the loop. That's how it works with graceful shutdown, for example. Let's leave it as is.

Reply:
Ok. Fixed.